### DISCUSSION TITLE
Vitest envirenment issues/settings - API testing

### DISCUSSION THREAD

**Discussion by oleksii-petrashchuk** [vitest-dev/vitest#3803] [Q&A]
### Issue
During POC with using the vite, I faced some issues with tests. For the API requests the _**ky**_ ([link](https://www.npmjs.com/package/ky)) is being used. When running tests, there was an issue with **_globalThis.Headers_**. After some digging into I found out that by default **_Node_** environment is used. So I set the _**jsdom**_ env to be in place. The issue with _**globalThis.Headers**_ is gone, but there is one more. 

The issue at the moment is still with environment (most probably). It says:
`TypeError: Cannot read property 'bind' of undefined`

Looking into stack trace there is a thing in the _**ky**_ library source code where it fails
`fetch: options.fetch ?? globalThis.fetch.bind(globalThis),`

So it appears that _**globalThis.fetch**_ is not defined, which means either of 2 things:
1.  Wrong env is still being used.
2.  Ky library can't be used for such a case.

Can someone assist with finding the root cause of the issue and possible fix?

### Settings
setupTests.ts
```
// jest-dom adds custom jest matchers for asserting on DOM nodes.
// allows you to do things like:
// expect(element).toHaveTextContent(/react/i)
// learn more: https://github.com/testing-library/jest-dom
import "@testing-library/jest-dom";

globalThis.IS_REACT_ACT_ENVIRONMENT = true;

```

vite.config.ts
```
/// <reference types="vitest" />\n
import { defineConfig } from 'vitest/config';
import react from '@vitejs/plugin-react';
import eslint from 'vite-plugin-eslint';

// https://vitejs.dev/config/
export default defineConfig({
  base: "/folder",
  plugins: [react(), eslint()],
  resolve: {
    alias: {
      src: "/src",
    },
  },
  test: {
    environment: "jsdom",
    setupFiles: "./setupTests.ts",
    globals: true,
  },
});
```

package.json
 ```
 "devDependencies": {
    "@testing-library/cypress": "^9.0.0",
    "@testing-library/jest-dom": "^5.16.5",
    "@testing-library/react": "^14.0.0",
    "@testing-library/react-hooks": "^8.0.1",
    "@testing-library/user-event": "^14.4.3",
    "@types/lodash": "^4.14.195",
    "@types/react": "^18.2.9",
    "@types/react-dom": "^18.0.11",
    "@typescript-eslint/eslint-plugin": "^5.59.9",
    "@typescript-eslint/parser": "^5.59.9",
    "@vitejs/plugin-react": "^4.0.0",
    "cypress": "^12.14.0",
    "env-cmd": "^10.1.0",
    "eslint": "^8.42.0",
    "eslint-config-prettier": "^8.8.0",
    "eslint-config-react-app": "^7.0.1",
    "eslint-plugin-cypress": "^2.13.3",
    "eslint-plugin-formatjs": "^4.10.3",
    "eslint-plugin-react": "^7.32.2",
    "eslint-plugin-react-hooks": "^4.6.0",
    "eslint-plugin-react-refresh": "^0.3.4",
    "jsdom": "^22.1.0",
    "junit": "^1.4.9",
    "msw": "^1.2.2",
    "prettier": "^2.8.8",
    "sass": "^1.63.2",
    "start-server-and-test": "^2.0.0",
    "typescript": "^5.0.2",
    "vite": "^4.3.9",
    "vite-plugin-eslint": "^1.8.1",
    "vitest": "^0.31.4"
  }
```

**Comment by sheremet-va** [comment]
Use Node.js version that has fetch implemented or install fetch polyfill in setup files
